zio-http-datastar-sdk/src/main/scala/zio/http/datastar/ServerSentEventGenerator.scala

```diff
@@ -0,0 +1,195 @@
+    )
+  }
+
+  def patchElements(
+    elements: String,
+    options: PatchElementOptions,
+  ): ZIO[Datastar, Nothing, Unit] =
+    patchElements(elements.split('\n').map(Dom.raw).toList, options)
+
+  def patchElements(
+    elements: String,
+  ): ZIO[Datastar, Nothing, Unit] =
+    patchElements(elements, PatchElementOptions.default)
+
+  def patchElements(
+    element: Dom,
+    options: PatchElementOptions,
+  ): ZIO[Datastar, Nothing, Unit] = patchElements(Iterable(element), options)
+
+  def patchElements(
+    element: Dom,
+  ): ZIO[Datastar, Nothing, Unit] =
+    patchElements(Iterable(element), PatchElementOptions.default)
+
+  def patchElements(
+    elements: Iterable[Dom],
+  ): ZIO[Datastar, Nothing, Unit] =
+    patchElements(elements, PatchElementOptions.default)
+
+  def patchElements(
+    elements: Iterable[Dom],
+    options: PatchElementOptions,
+  ): ZIO[Datastar, Nothing, Unit] = {
+
+    val sb = new StringBuilder
+
+    options.selector.foreach(s => sb.append("selector ").append(s.render).append("\n"))
+
+    if (options.mode != ElementPatchMode.Outer) {
+      sb.append("mode ").append(options.mode.toString).append("\n")
+    }
+
+    if (options.useViewTransition) {
+      sb.append("useViewTransition true\n")
+    }
+
+    elements.foreach(d => {
+      sb.append("elements ").append(d.render).append("\n")
+    })
+
+    val retry = if (options.retryDuration != 1000.millis) Some(options.retryDuration) else None
+    send(EventType.PatchElements, sb.toString(), options.eventId, retry)
+  }
+
+  def patchSignals(
+    signal: String,
+    options: PatchSignalOptions,
+  ): ZIO[Datastar, Nothing, Unit] =
+    patchSignals(Iterable(signal), options)
+
+  def patchSignals(
+    signal: String,
+  ): ZIO[Datastar, Nothing, Unit] =
+    patchSignals(Iterable(signal), PatchSignalOptions.default)
+
+  def patchSignals(
+    signals: Iterable[String],
+  ): ZIO[Datastar, Nothing, Unit] =
+    patchSignals(signals, PatchSignalOptions.default)
+
+  def patchSignals(
+    signals: Iterable[String],
+    options: PatchSignalOptions,
+  ): ZIO[Datastar, Nothing, Unit] = {
+
+    val sb = new StringBuilder
+
+    if (options.onlyIfMissing) {
+      sb.append("onlyIfMissing true\n")
+    }
+
+    signals.foreach(s => {
+      sb.append("signals ").append(s).append("\n")
+    })
+
+    val retry = if (options.retryDuration != 1000.millis) Some(options.retryDuration) else None
+    send(EventType.PatchSignals, sb.toString(), options.eventId, retry)
+  }
+
+  private def send(
+    eventType: EventType,
+    dataLines: String,
+    eventId: Option[String] = None,
+    retry: Option[Duration] = Some(1000.millis),
+  ): ZIO[Datastar, Nothing, Unit] =
+    ZIO.serviceWithZIO[Datastar] { `d*` =>
+      val retry0 = if (retry.contains(1000.millis)) None else retry
+      `d*`.queue
+        .offer(ServerSentEvent(data = dataLines, eventType = Some(eventType.toString), id = eventId, retry = retry0))
```

Comment:
See https://github.com/zio/zio-http/pull/3649/files#r2297080439

```suggestion
        .offer(ServerSentEvent(data = dataLines, eventType = Some(eventType.render), id = eventId, retry = retry0))
```